### DISCUSSION TITLE
Error 404

### DISCUSSION THREAD

**Discussion by PrieserMax** [YOURLS/YOURLS#3175] [Q&A] [answered]
Hello,

I hope I'm right here.

I have today setup yourLS and (hopefully) configured correctly and also created the .htaccess file with the following content according to the tutorial:

```
# BEGIN YOURLS
<IfModule mod_rewrite.c>
RewriteEngine On
RewriteBase /
RewriteCond %{REQUEST_FILENAME} !-f
RewriteCond %{REQUEST_FILENAME} !-d
RewriteRule ^.*$ /yourls-loader.php [L]
</IfModule>
# END YOURLS
```

My Config:

```
<?php
/* This is a sample config file.
 * Edit this file with your own settings and save it as "config.php"
 *
 * IMPORTANT: edit and save this file as plain ASCII text, using a text editor, for instance TextEdit on Mac OS or
 * Notepad on Windows. Make sure there is no character before the opening <?php at the beginning of this file.
 */

/*
 ** MySQL settings - You can get this info from your web host
 */

/** MySQL database username */
define( 'YOURLS_DB_USER', 'yourlsXXXX' );

/** MySQL database password */
define( 'YOURLS_DB_PASS', 'MyPassword' );

/** The name of the database for YOURLS
 ** Use lower case letters [a-z], digits [0-9] and underscores [_] only */
define( 'YOURLS_DB_NAME', 'dbMyDBName' );

/** MySQL hostname.
 ** If using a non standard port, specify it like 'hostname:port', e.g. 'localhost:9999' or '127.0.0.1:666' */
define( 'YOURLS_DB_HOST', 'MyHostersDomainForDBs' );

/** MySQL tables prefix
 ** YOURLS will create tables using this prefix (eg `yourls_url`, `yourls_options`, ...)
 ** Use lower case letters [a-z], digits [0-9] and underscores [_] only */
define( 'YOURLS_DB_PREFIX', 'yourls_' );

/*
 ** Site options
 */

/** YOURLS installation URL
 ** All lowercase, no trailing slash at the end.
 ** If you define it to "http://sho.rt", don't use "http://www.sho.rt" in your browser (and vice-versa)
 ** To use an IDN domain (eg http://héhé.com), write its ascii form here (eg http://xn--hh-bjab.com) */
define( 'YOURLS_SITE', 'https://MyDomain.xy' );

/** YOURLS language
 ** Change this setting to use a translation file for your language, instead of the default English.
 ** That translation file (a .mo file) must be installed in the user/language directory.
 ** See http://yourls.org/translations for more information */
define( 'YOURLS_LANG', 'de_DE' );

/** Allow multiple short URLs for a same long URL
 ** Set to true to have only one pair of shortURL/longURL (default YOURLS behavior)
 ** Set to false to allow multiple short URLs pointing to the same long URL (bit.ly behavior) */
define( 'YOURLS_UNIQUE_URLS', false );

/** Private means the Admin area will be protected with login/pass as defined below.
 ** Set to false for public usage (eg on a restricted intranet or for test setups)
 ** Read http://yourls.org/privatepublic for more details if you're unsure */
define( 'YOURLS_PRIVATE', false );

/** A random secret hash used to encrypt cookies. You don't have to remember it, make it long and complicated
 ** Hint: copy from http://yourls.org/cookie */
define( 'YOURLS_COOKIEKEY', 'CodeFromUpperProvidedUrl' );

/** Username(s) and password(s) allowed to access the site. Passwords either in plain text or as encrypted hashes
 ** YOURLS will auto encrypt plain text passwords in this file
 ** Read http://yourls.org/userpassword for more information */
$yourls_user_passwords = [
	'admin' => 'phpass:XXXYYYZZZ /* Password encrypted by YOURLS */ ,
	// 'username2' => 'password2',
	// You can have one or more 'login'=>'password' lines
];

/** URL shortening method: either 36 or 62
 ** 36: generates all lowercase keywords (ie: 13jkm)
 ** 62: generates mixed case keywords (ie: 13jKm or 13JKm) 
 ** For more information, see https://yourls.org/urlconvert */
define( 'YOURLS_URL_CONVERT', 36 );

/** Debug mode to output some internal information
 ** Default is false for live site. Enable when coding or before submitting a new issue */
define( 'YOURLS_DEBUG', true );

/**
* Reserved keywords (so that generated URLs won't match them)
* Define here negative, unwanted or potentially misleading keywords.
*/
$yourls_reserved_URL = [
	'Don'tWantToMentionHere'
];

/*
 ** Personal settings would go after here.
 */
```

I can also access the secured admin interface and created the public interface.

The website is also correctly configured with TLS/SSL.
My PHP-version selected on the hoster is 8.0, the MySQL version is 7.4. the yourLS version is the newest, 1.8.2 (the hoster is a german hoster "Strato").

The problem I have at the moment is that I can create yourLS short links, but I can't access them.
They also show up inside the database and the admin interface, but can't get accessed. I get the standard 404 error when trying to open them (404 Not Found The requested URL was not found on this server.)

What else could it be?

Greeting from South-Germany

Max

**Comment by dgw** [comment, marked as answer]
My first step would be to make sure (via documentation or by contacting the helpdesk) that the host 0) actually uses the Apache server, 1) allows `.htaccess` files, 2) has enabled the required `mod_rewrite.c` module, and 3) allows said module to be invoked inside `.htaccess` files. Beyond that, please check your server logs as suggested in our dedicated [wiki page about 404 errors](https://docs.yourls.org/guide/troubleshooting/frequent-issues.html#_404-instead-of-redirection) and follow the [troubleshooting steps](https://docs.yourls.org/guide/troubleshooting/first-steps.html) as much as you can.

This is a fairly common issue, and it's quite likely you can find someone else who even experienced it on the same webhost you use.

**Comment by PrieserMax** [comment]
Hey, I appreciate your help.
I found out the source of the problem, it was me.
I didn't know, I would have to enter http:// / https:// in front of the URL.

But thank you for your help.

(BTW. I tried the troubleshooting steps, but it didn't work, because I'm dumb xD)

**Reply by tkredbaron** [reply]
hey,
i´m not be able to solve my problem.
i install in a subdirectory and today the first day some call and told its not working. i look in logfile and found:

AH01276: Cannot serve directory /var/www/vhosts/mydomain.com/httpdocs/review/: No matching DirectoryIndex (index.html,index.cgi,index.pl,index.php,index.xhtml,index.htm,index.shtml) found, and server-generated directory index forbidden by Options directive
--

client has windows edge browser and nokia XR20 mobile.

.htaccess

# BEGIN YOURLS
<IfModule mod_rewrite.c>
RewriteEngine On
RewriteBase /review/
RewriteCond %{REQUEST_FILENAME} !-f
RewriteCond %{REQUEST_FILENAME} !-d
RewriteRule ^.*$ /review/yourls-loader.php [L]
</IfModule>
# END YOURLS

yourls running under mydomain.com/review/
why error shows 'mydomain.com/httpdocs/review/'

kindly regards

tobi
